drivers/counter/counter_ll_stm32_timer.c

```diff
@@ -0,0 +1,632 @@
+		apb_psc = STM32_D2PPRE1;
+	} else {
+		apb_psc = STM32_D2PPRE2;
+	}
+#else
+	if (pclken->bus == STM32_CLOCK_BUS_APB1) {
+		apb_psc = STM32_APB1_PRESCALER;
+	}
+#if !defined(CONFIG_SOC_SERIES_STM32F0X) && !defined(CONFIG_SOC_SERIES_STM32G0X)
+	else {
+		apb_psc = STM32_APB2_PRESCALER;
+	}
+#endif
+#endif
+
+#if defined(RCC_DCKCFGR_TIMPRE) || defined(RCC_DCKCFGR1_TIMPRE) || \
+	defined(RCC_CFGR_TIMPRE)
+	/*
+	 * There are certain series (some F4, F7 and H7) that have the TIMPRE
+	 * bit to control the clock frequency of all the timers connected to
+	 * APB1 and APB2 domains.
+	 *
+	 * Up to a certain threshold value of APB{1,2} prescaler, timer clock
+	 * equals to HCLK. This threshold value depends on TIMPRE setting
+	 * (2 if TIMPRE=0, 4 if TIMPRE=1). Above threshold, timer clock is set
+	 * to a multiple of the APB domain clock PCLK{1,2} (2 if TIMPRE=0, 4 if
+	 * TIMPRE=1).
+	 */
+
+	if (LL_RCC_GetTIMPrescaler() == LL_RCC_TIM_PRESCALER_TWICE) {
+		/* TIMPRE = 0 */
+		if (apb_psc <= 2u) {
+			LL_RCC_ClocksTypeDef clocks;
+
+			LL_RCC_GetSystemClocksFreq(&clocks);
+			*tim_clk = clocks.HCLK_Frequency;
+		} else {
+			*tim_clk = bus_clk * 2u;
+		}
+	} else {
+		/* TIMPRE = 1 */
+		if (apb_psc <= 4u) {
+			LL_RCC_ClocksTypeDef clocks;
+
+			LL_RCC_GetSystemClocksFreq(&clocks);
+			*tim_clk = clocks.HCLK_Frequency;
+		} else {
+			*tim_clk = bus_clk * 4u;
+		}
+	}
+#else
+	/*
+	 * If the APB prescaler equals 1, the timer clock frequencies
+	 * are set to the same frequency as that of the APB domain.
+	 * Otherwise, they are set to twice (×2) the frequency of the
+	 * APB domain.
+	 */
+	if (apb_psc == 1u) {
+		*tim_clk = bus_clk;
+	} else {
+		*tim_clk = bus_clk * 2u;
+	}
+#endif
+
+	return 0;
+}
+
+static int init_timer(const struct device *dev)
+{
+	struct counter_stm32_config *cfg =
+		(struct counter_stm32_config *)get_stm32_config(dev);
+	TIM_TypeDef *timer = cfg->timer;
+	LL_TIM_InitTypeDef init;
+	uint32_t tim_clk;
+	int r;
+
+	/* initialize clock and check its speed  */
+	r = clock_control_on(DEVICE_DT_GET(STM32_CLOCK_CONTROL_NODE),
+			     (clock_control_subsys_t *)&cfg->pclken);
+	if (r < 0) {
+		LOG_ERR("Could not initialize clock (%d)", r);
+		return r;
+	}
+	r = get_tim_clk(&cfg->pclken, &tim_clk);
+	if (r < 0) {
+		LOG_ERR("Could not obtain timer clock (%d)", r);
+		return r;
+	}
+	cfg->info.freq = tim_clk / (cfg->prescaler + 1);
+
+	/* enable IRQ */
+	NVIC_SetPriority(cfg->irqn, 0);
+	NVIC_EnableIRQ(cfg->irqn);
+
+	/* initialize timer */
+	LL_TIM_StructInit(&init);
+
+	init.Prescaler = cfg->prescaler;
+	init.CounterMode = LL_TIM_COUNTERMODE_UP;
+	init.Autoreload = counter_get_max_top_value(dev);
+	init.ClockDivision = LL_TIM_CLOCKDIVISION_DIV1;
+
+	if (LL_TIM_Init(timer, &init) != SUCCESS) {
+		LOG_ERR("Could not initialize timer");
+		return -EIO;
+	}
+
+	return 0;
+}
+
+static uint32_t get_guard_period(const struct device *dev, uint32_t flags)
+{
+	return get_dev_data(dev)->guard_period;
+}
```

Comment:
What's the purpose of `flags` arg ?
Besides, this function looks superfluous

Reply:
This function is part of the counter API, so I believe the arguments are standardized?
```
	.get_guard_period = get_guard_period,
```

Reply:
ok, then please add the following line:
```
ARG_UNUSED(flags);
```

Reply:
Gotcha, done.